charts/delphai-deployment/templates/deployment.yaml

```diff
@@ -202,12 +202,44 @@ spec:
 
           ports:
             {{ range $portName, $portConf := $.Values.ports }}
+            {{ if ne $portName "oauthproxy" }}
             {{ if or (eq $portConf.enabled nil) $portConf.enabled }}
             - name: {{ $portName | quote }}
               containerPort: {{ $portConf.port }}
             {{ end }} {{/* if or (eq $portConf.enabled nil) $portConf.enabled */}}
+            {{ end }} {{/* if ne $portName "oauthproxy" */}}
             {{ end }} {{/* range ports */}}
 
+          {{ if or (eq $.Values.readinessProbe.enabled nil) $.Values.readinessProbe.enabled }}
+
+          {{/* readinessProbe.portName input validation/open port discovery */}}
+          {{ $readinessProbePortName := $.Values.readinessProbe.portName }}
+          {{ if eq $readinessProbePortName "oauthproxy" }}
+            {{ fail "readinessProbe.portName must not be 'oauthproxy'." }}
+          {{ else if eq $readinessProbePortName "" }}
+            {{ range $portName, $portConf := $.Values.ports }}
+            {{ if eq $portName "oauthproxy" }}
+              {{ continue }}
+            {{ end }} {{/* if eq $portName "oauthproxy" */}}
+            {{ if or (eq $portConf.enabled nil) $portConf.enabled }}
+              {{ $readinessProbePortName = $portName }}
+              {{ break }} {{/* range $portName, $portConf := $.Values.ports */}}
+            {{ end }} {{/* if or (eq $portConf.enabled nil) $portConf.enabled */}}
+            {{ end }} {{/* range $portName, $portConf := $.Values.ports */}}
+          {{ else if not (index $.Values.ports $readinessProbePortName).enabled }}
+            {{ fail "readinessProbe.portName must correspond to an enabled port." }}
+          {{ end }} {{/* if */}}
+
+          {{ if ne $readinessProbePortName "" }}
+          readinessProbe:
+            tcpSocket:
+              port: {{ $readinessProbePortName }}
+            initialDelaySeconds: {{ $.Values.readinessProbe.initialDelaySeconds }}
+            periodSeconds: {{ $.Values.readinessProbe.periodSeconds }}
+            timeoutSeconds: {{ $.Values.readinessProbe.timeoutSeconds }}
```

Comment:
You can use maybe
```suggestion
            {{- omit $.Values.readinessProbe "enabled" | toYaml | nindent 14 }}
```

Reply:
Yes sure, just showed the idea

Reply:
Also not sure if it's 14 or some other number

Reply:
committed, fixed nindent, and tested